pkg/oidc/oidc.go

```diff
@@ -91,6 +118,11 @@ func (a *Authenticator) generateCodeVerifier(len int) (string, string) {
 
 // Authenticate with IDP
 func (a *Authenticator) Authenticate(ctx context.Context) (payload string, err error) {
+	a.template, err = template.New("response").Parse(responseTpl)
```

Comment:
iirc there is a `template.Must` for doing this (and panic if it fails, which is appropriate, as it is a static template)

Reply:
This `Authenticate` function is already returning error, do we want to use panic in this case?

Reply:
If the static template fails to parse, I think a panic is appropriate. We should also parse it at startup, so we panic early.

Reply:
I'm not sure I agree panic is appropriate here. There is error handling in place, and parsing failure is probably a common error, not so exceptional that worth panicking. But I do agree it's a good idea to move parsing logic outside of this function, since this API is for authentication, not template parsing.

Reply:
We should *parse* the template at startup and panic if it fails to parse. We should evaluate that parsed template here. If the evaluation of it fails, then error is fine. If the parsing fails, then it should panic at startup.

Reply:
The only reason *parsing* the template would fail is if the template is broken. This means it will *never* parse, as it is a a static string.